Microsoft Academic.js

```diff
@@ -0,0 +1,426 @@
+        if (checkOnly) return true;
+        found = true;
+        items[href] = title;
+    }
+    return found ? items : false;
+}
+
+
+function doWeb(doc, url) {
+    if (detectWeb(doc, url) == "multiple") {
+        Zotero.selectItems(getSearchResults(doc, false), function (items) {
+            if (!items) {
+                return true;
+            }
+            var articles = [];
+            for (var i in items) {
+                articles.push(i);
+            }
+            ZU.processDocuments(articles, scrape);
+        });
+    } else {
+        scrape(doc, url);
+    }
+}
+
+
+function scrape(doc, url) {
+	var pubID = url.match(/\/detail\/(\d+)/)[1];
+	
+	var apiUrl = "https://academic.microsoft.com/api/browse/GetEntityDetails?entityId=" +
+		pubID + "&correlationId=undefined";
+	
+	ZU.doGet(apiUrl, function(text) {
+		var data = JSON.parse(text);
+		var type = detectWeb(doc, url);
+		var item = new Zotero.Item(type);
+		item.itemID = pubID;
+		item.title = data.entityTitle;
+		item.date = data.entity.d;//alternatively ZU.strToISO(data.date);
+		item.abstractNote = data.abstract;
+		
+		if (data.authors) {
+			for (var i=0; i<data.authors.length; i++) {
+				item.creators.push(ZU.cleanAuthor(data.authors[i].lt, "author"));
+			}
+		}
+
+		item.publicationTitle = data.entity.extended.vfn;
+		item.volume = data.entity.extended.v;
+		item.issue = data.entity.extended.i;
+		item.pages = data.entity.extended.fp;
+		if (data.entity.extended.lp) {
+			item.pages += "–" + data.entity.extended.lp;
+		}
+		item.DOI = data.entity.extended.doi;
+		
+		if (data.fieldsOfStudy) {
+			for (var i=0; i<data.fieldsOfStudy.length; i++) {
+				item.tags.push(data.fieldsOfStudy[i].lt);
+			}
+		}
+		
+		item.attachments.push({
+			title: "Snapshot",
+			document: doc
+		});
+		if (data.sources) {
+			for (var i=0; i<data.sources.length; i++) {
+				item.attachments.push({
+					title: "Source",
+					url: data.sources[i].u,
+					snapshot: false
+				});
+			}
+		}
+		
+		/*
+		delete data.references;
+		delete data.citations;
+		Z.debug(data);
+		*/
+		
+		item.complete();
+	});
+	
+}
+/** BEGIN TEST CASES **/
+var testCases = [
+	{
+		"type": "web",
+		"url": "https://academic.microsoft.com/#/detail/2084324324",
+		"items": [
+			{
+				"itemType": "journalArticle",
+				"title": "Out of Cite! How Reference Managers Are Taking Research to the Next Level",
+				"creators": [
+					{
+						"firstName": "Jason",
+						"lastName": "Muldrow",
+						"creatorType": "author"
+					},
+					{
+						"firstName": "Stephen",
+						"lastName": "Yoder",
+						"creatorType": "author"
+					}
+				],
+				"date": "2009-01-01",
+				"DOI": "10.1017/S1049096509090337",
+				"issue": 1,
+				"itemID": "2084324324",
+				"libraryCatalog": "Microsoft Academic",
+				"pages": "167–172",
+				"publicationTitle": "PS Political Science & Politics",
+				"volume": 42,
+				"attachments": [
+					{
+						"title": "Snapshot"
+					},
+					{
```

Comment:
I think putting the links wrapped into a href= into a note makes sense (and they'd be clickable that way). I see that they're useful, but  I don't think users will be pleased with 10+ links for items.